### PR TITLE
Add enableStripePromoCodes config

### LINKED ISSUE #12111: Enable Stripe promo codes via config
Stripe have released a feature for [applying promo codes to subscriptions](https://twitter.com/stripe/status/1291441519585406976).

Ghost should expose (via config) an option to enable this functionality in Stripe. There's an [open PR](https://github.com/TryGhost/Members/pull/194) to enable this in Members, but the value is not taken from config.

The config flag (and the `allow_promotion_codes` flag passed to Stripe) should be disabled by default.

### PR DESCRIPTION
**Pull request by kristianfreeman**
This PR adds support for reading the enableStripePromoCodes config option, which can be used in tandem with the work at https://github.com/TryGhost/Members/pull/194 to enable Ghost blogs to turn on support for Stripe promotional codes during the member checkout process.

Closes #12111

### CODE REVIEW

File: core/server/services/members/config.js [hunk 1]
```diff
@@ -154,6 +154,10 @@ class MembersConfigProvider {
         };
     }
 
+    getEnableStripePromoCodes() {
+        return this._config.get('enableStripePromoCodes');
```
Comment (signalnerve) [inline core/server/services/members/config.js:158]: Is this an OK way to accept that config input? Just wondering if there's a place where nconf validates this as a boolean, or if I should just do something like `this._config.get('enableStripePromoCodes') === true` instead

**Comment by signalnerve** [comment]
hey all, any thoughts on this PR? would love to help get this over the finish line so i can use it my own sites :)

**Comment by allouis** [comment]
Hey @signalnerve apologies for the delay! 

This looks great and the right direction, I only have one comment though - it seems to be that `enableStripePromoCodes` config would be better placed _inside_ of the `stripe` config, rather than as a separate item, e.g.

```js
paymentConfig: {
  stripe: {
    enablePromoCodes: Boolean,
    // the rest of the stripe related config
  }
}
```

File: core/server/services/members/config.js (hunk 1 above)
Comment (allouis) [reply core/server/services/members/config.js:158]: This looks good to me :relaxed:

**Comment by signalnerve** [comment]
Got it, thanks @allouis! I just updated this PR (https://github.com/TryGhost/Ghost/pull/12149/commits/4fc32498c8d1cbf210bab8babcd4e48e897291d5) and the associated PR in the Member repo (https://github.com/TryGhost/Members/pull/194/commits/df704a54dcebaaa8476957864093f35e7c8626c1) to embed this in the Stripe config. 100% agree on that idea, it's way cleaner like this. Let me know what you think!

**Comment by signalnerve** [comment]
Hey @allouis, bump on this! Anything I can do to help get review on this and my other PR?

**Comment by ErisDS** [comment]
Hey, a lot of our team are currently on leave (it's that time of year!) we'll get these reviewed and merged as soon as we can, but there may be a delay.

**Comment by allouis** [comment]
Hey @signalnerve - thanks for your patience!

This looks good :) The Members PR matches up well too!

I think there's just one outstanding task for this which is to add adefault value for the config here https://github.com/TryGhost/Ghost/blob/master/core/shared/config/defaults.json (should be false to preserve current functionality)

**Comment by signalnerve** [comment]
Got it! @allouis just pushed that change 👍

**Comment by allouis** [comment]
Thank you @signalnerve!! :tada:

**Comment by allouis** [comment]
This has been released in [3.34.0](https://github.com/TryGhost/Ghost/releases/tag/3.34.0) :woman_dancing:

**Comment by signalnerve** [comment]
sweet, thanks @allouis! is there plans to include opting into this feature in the documentation?

**Comment by kmskrishna** [comment]
Hey @signalnerve I couldn't find this in the documentation yet. Am I missing something?

**Comment by boxedpaul** [comment]
Bump...when we will be able to enable this feature in the Portal Settings? (suggested implementation)

image

**Comment by signalnerve** [comment]
@kmskrishna @boxedpaul I'm not on the team so I can't speak to plans on integrating this in the UI 🤷  it's super useful so I hope that happens soon!

**Comment by jeffreycunn** [comment]
Any updates on this feature being added into Ghost? I think this will be super useful.

**Pull request merged**